codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -272,6 +275,13 @@ private boolean readClientPrefaceString(ByteBuf in) throws Http2Exception {
             if (bytesRead == 0 || !ByteBufUtil.equals(in, in.readerIndex(),
                                                       clientPrefaceString, clientPrefaceString.readerIndex(),
                                                       bytesRead)) {
+                String http1Request =
```

Comment:
I tried adding a method to ByteBufUtil.  This skips the parsing to ascii, though, I think it gives up any sort of smart string search the JVM would do.